### PR TITLE
Make CI compile aiohttp again.

### PR DESCRIPTION
**Pull request by xavierfiechter**
References: 
https://stackoverflow.com/a/64755338/20124004,
https://stackoverflow.com/a/64755052/20124004, 
https://github.com/kivy/python-for-android/pull/2518

### CODE REVIEW

File: pythonforandroid/recipes/aiohttp/__init__.py [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+from pythonforandroid.recipe import CppCompiledComponentsPythonRecipe
 
-
-class AIOHTTPRecipe(CythonRecipe):  # type: ignore # pylint: disable=R0903
-    """Build AIOHTTP"""
-
-    version = "v3.6.2"
-    url = "https://github.com/aio-libs/aiohttp/archive/{version}.zip"
+class AIOHTTPRecipe(CppCompiledComponentsPythonRecipe):  # type: ignore # pylint: disable=R0903
+    url = "https://files.pythonhosted.org/packages/ff/4f/62d9859b7d4e6dc32feda67815c5f5ab4421e6909e48cbc970b6a40d60b7/aiohttp-3.8.3.tar.gz"
```
Comment (AndreMiras) [inline pythonforandroid/recipes/aiohttp/__init__.py:6]: Can you update that line to leverage the version variable, and also use the github url?
Like so:
```python
version = "3.8.3"
url = "https://github.com/aio-libs/aiohttp/archive/v{version}.zip"
```

**Review by AndreMiras — COMMENTED** [review]
Thanks for the PR, I left a comment regarding the recipe URL

**Review by AndreMiras — DISMISSED** [review]
Thanks Xavier, I'll merge when the CI turns green for this recipe

File: pythonforandroid/recipes/aiohttp/__init__.py [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
     name = "aiohttp"
-
     depends: List[str] = ["setuptools"]
+    call_hostpython_via_targetpython = False
+    install_in_hostpython = True
 
+    def get_recipe_env(self, arch):
+        env = super().get_recipe_env(arch)
+        env['LDFLAGS'] += ' -lc++_shared'
+        return env
```
Comment (AndreMiras) [inline pythonforandroid/recipes/aiohttp/__init__.py:18]: Flake8 is expecting an extra line here, CI is complaining:
```
pythonforandroid/recipes/aiohttp/__init__.py:19:1: E305 expected 2 blank lines after class or function definition, found 1
```

**Comment by AndreMiras** [comment]
It seems like the recipe failed in the CI with the following:
```
stty: stdin isn't a terminal
[INFO]:    -> running python3 setup.py clean --all
stty: stdin isn't a terminal
[INFO]:    -> running python3 setup.py build_ext -v
Exception in thread background thread for pid 77842:
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.10/3.10.7/Frameworks/Python.framework/Versions/3.10/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python@3.10/3.10.7/Frameworks/Python.framework/Versions/3.10/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/runner/work/python-for-android/python-for-android/venv/lib/python3.10/site-packages/sh.py", line 1641, in wrap
    fn(*rgs, **kwargs)
  File "/Users/runner/work/python-for-android/python-for-android/venv/lib/python3.10/site-packages/sh.py", line 2569, in background_thread
    handle_exit_code(exit_code)
  File "/Users/runner/work/python-for-android/python-for-android/venv/lib/python3.10/site-packages/sh.py", line [2269](https://github.com/kivy/python-for-android/actions/runs/3281492031/jobs/5404298939#step:6:2270), in fn
    return self.command.handle_command_exit_code(exit_code)
  File "/Users/runner/work/python-for-android/python-for-android/venv/lib/python3.10/site-packages/sh.py", line 869, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_1: 

  RAN: /Users/runner/.python-for-android/build/other_builds/hostpython3/desktop/hostpython3/native-build/python3 setup.py build_ext -v

  STDOUT:
*********************
* Accelerated build *
*********************
running build_ext
building 'aiohttp._websocket' extension
creating build
creating build/temp.macosx-11.7-x86_64-3.9
creating build/temp.macosx-11.7-x86_64-3.9/aiohttp
/Users/runner/.android/android-ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang -target armv7a-linux-androideabi21 -fomit-frame-pointer -march=armv7-a -mfloat-abi=softfp -mfpu=vfp -mthumb -fPIC -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -target armv7a-linux-androideabi21 -fomit-frame-pointer -march=armv7-a -mfloat-abi=softfp -mfpu=vfp -mthumb -fPIC -I/Users/runner/.python-for-android/build/other_builds/python3/armeabi-v7a__ndk_target_21/python3/Include -DANDROID -I/Users/runner/.android/android-ndk/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/include -I/Users/runner/.python-for-android/build/python-installs/bdist_unit_tests_app/armeabi-v7a/include/python3.9 -I/Users/runner/.python-for-android/build/other_builds/hostpython3/desktop/hostpython3/Include -I/Users/runner/.python-for-android/build/other_builds/hostpython3/desktop/hostpython3/native-build -c aiohttp/_websocket.c -o build/temp.macosx-11.7-x86_64-3.9/aiohttp/_websocket.o
clang-14: error: no such file or directory: 'aiohttp/_websocket.c'
clang-14: error: no input files
error: command '/Users/runner/.android/android-ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang' failed with exit code 1


  STDERR:

Traceback (most recent call last):
  File "/Users/runner/work/python-for-android/python-for-android/testapps/on_device_unit_tests/setup.py", line 106, in <module>
    setup(
  File "/Users/runner/work/python-for-android/python-for-android/venv/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 185, in setup
    return run_commands(dist)
  File "/Users/runner/work/python-for-android/python-for-android/venv/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
    dist.run_commands()
  File "/Users/runner/work/python-for-android/python-for-android/venv/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 973, in run_commands
    self.run_command(cmd)
  File "/Users/runner/work/python-for-android/python-for-android/venv/lib/python3.10/site-packages/setuptools/dist.py", line 1217, in run_command
    super().run_command(command)
  File "/Users/runner/work/python-for-android/python-for-android/venv/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 992, in run_command
    cmd_obj.run()
  File "/Users/runner/work/python-for-android/python-for-android/pythonforandroid/bdistapk.py", line 79, in run
    main()
  File "/Users/runner/work/python-for-android/python-for-android/pythonforandroid/entrypoints.py", line 18, in main
    ToolchainCL()
  File "/Users/runner/work/python-for-android/python-for-android/pythonforandroid/toolchain.py", line 730, in __init__
    getattr(self, command)(args)
  File "/Users/runner/work/python-for-android/python-for-android/pythonforandroid/toolchain.py", line 153, in wrapper_func
    build_dist_from_args(ctx, dist, args)
  File "/Users/runner/work/python-for-android/python-for-android/pythonforandroid/toolchain.py", line 212, in build_dist_from_args
    build_recipes(build_order, python_modules, ctx,
  File "/Users/runner/work/python-for-android/python-for-android/pythonforandroid/build.py", line 505, in build_recipes
    recipe.build_arch(arch)
  File "/Users/runner/work/python-for-android/python-for-android/pythonforandroid/recipe.py", line 983, in build_arch
    self.build_compiled_components(arch)
  File "/Users/runner/work/python-for-android/python-for-android/pythonforandroid/recipe.py", line 994, in build_compiled_components
    shprint(hostpython, 'setup.py', self.build_cmd, '-v',
  File "/Users/runner/work/python-for-android/python-for-android/pythonforandroid/logger.py", line 167, in shprint
    for line in output:
  File "/Users/runner/work/python-for-android/python-for-android/venv/lib/python3.10/site-packages/sh.py", line 915, in next
    self.wait()
  File "/Users/runner/work/python-for-android/python-for-android/venv/lib/python3.10/site-packages/sh.py", line 845, in wait
    self.handle_command_exit_code(exit_code)
  File "/Users/runner/work/python-for-android/python-for-android/venv/lib/python3.10/site-packages/sh.py", line 869, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_1: 

  RAN: /Users/runner/.python-for-android/build/other_builds/hostpython3/desktop/hostpython3/native-build/python3 setup.py build_ext -v

  STDOUT:
*********************
* Accelerated build *
*********************
running build_ext
building 'aiohttp._websocket' extension
creating build
creating build/temp.macosx-11.7-x86_64-3.9
creating build/temp.macosx-11.7-x86_64-3.9/aiohttp
/Users/runner/.android/android-ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang -target armv7a-linux-androideabi21 -fomit-frame-pointer -march=armv7-a -mfloat-abi=softfp -mfpu=vfp -mthumb -fPIC -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -target armv7a-linux-androideabi21 -fomit-frame-pointer -march=armv7-a -mfloat-abi=softfp -mfpu=vfp -mthumb -fPIC -I/Users/runner/.python-for-android/build/other_builds/python3/armeabi-v7a__ndk_target_21/python3/Include -DANDROID -I/Users/runner/.android/android-ndk/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/include -I/Users/runner/.python-for-android/build/python-installs/bdist_unit_tests_app/armeabi-v7a/include/python3.9 -I/Users/runner/.python-for-android/build/other_builds/hostpython3/desktop/hostpython3/Include -I/Users/runner/.python-for-android/build/other_builds/hostpython3/desktop/hostpython3/native-build -c aiohttp/_websocket.c -o build/temp.macosx-11.7-x86_64-3.9/aiohttp/_websocket.o
clang-14: error: no such file or directory: 'aiohttp/_websocket.c'
clang-14: error: no input files
error: command '/Users/runner/.android/android-ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang' failed with exit code 1
```
See for instance this job: https://github.com/kivy/python-for-android/actions/runs/3281492031/jobs/5404298939
Do you have an idea why?

**Comment by xavierfiechter** [comment]
That's why I had the tarball from PyPI in the first PR. 

https://github.com/aio-libs/aiohttp/issues/3907#issuecomment-511767428

Can revert to the PyPI URL if you want me to do.

**Comment by AndreMiras** [comment]
Oh right I didn't know that, thanks for the heads up!
Then there's a world in between, try with this pattern instead:
```
url = "https://pypi.python.org/packages/source/a/aiohttp/aiohttp-{version}.tar.gz"
```

**Comment by xavierfiechter** [comment]
Even better.

**Review by AndreMiras — APPROVED** [review]
The recipe built successfully, thanks again for the pull request and follow up changes

**Comment by xavierfiechter** [comment]
@AndreMiras  I have two other custom baked recipes that could be streamlined. 

One needs a specific commit at the moment.

aiosocks at commit d4a85e73c9e3beadd7ab4c46b8f3ceb3b57338b2

Should I set commit instead of version

e.g. like this?
```
commit = d4a85e73c9e3beadd7ab4c46b8f3ceb3b57338b2
url = "git+https://github.com/nibrag/aiosocks.git@{commit}"
```

**Pull request merged**